src/ert/job_queue/job_queue_node.py

```diff
@@ -394,3 +473,26 @@ def wait_for(self) -> None:
 
     def _set_thread_status(self, new_status: ThreadStatus) -> None:
         self._thread_status = new_status
+
+
+class ProcessWithException(mp.Process):
+    """Used to run something in a subprocess, and capture exceptions. In order to catch
+    an exception, wait_and_throw_if_exception should be tried - before one joins the
+    process!"""
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        mp.Process.__init__(self, *args, **kwargs)
+        self._parent_connection, self._child_connection = mp.Pipe(False)
+        self._exception = None
+
+    def run(self) -> None:
+        try:
+            mp.Process.run(self)
+            self._child_connection.send(None)
```

Comment:
`run` of `ProcessWithException` is not called in tests.

Reply:
It should be called in tests, but it may not be picked up by coverage due to being run in a different process?